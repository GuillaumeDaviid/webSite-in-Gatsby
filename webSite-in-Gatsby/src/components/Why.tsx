import * as React from 'react'
import './styles/Why.scss'
import { StaticImage } from 'gatsby-plugin-image'


export default function Why() {

    return(
        <div className="Why">

            <h2 data-testid="title">POURQUOI MOI ?</h2>
            <div className="Why_separator" data-testid="separator"></div>
            <p className="Why_head" data-testid="text">Envie d&apos;un nouveau site web ? Ou alors d&apos;une refonte de celui-ci ? 
            Vous cherchez un développeur qui répond à vos besoins sur-mesure et qui respect les délais et le budget ?</p>
       

            <div className="Why_items">
                <div className="Why_items-elt" data-testid="items">
                    <div className="image_container">

                    </div>
                    <StaticImage
                        alt="Smartphone responsive"
                        src='../images/app.png'
                    />
                    <h3 className="items-title">Responsive Design</h3>
                    <p className="items-text">Site web accessible sur mobile.</p>
                </div>

                <div className="Why_items-elt" data-testid="items">
                    <div className="image_container">
              
                    </div>
                    <StaticImage
                        alt="Smartphone responsive"
                        src='../images/app2.png'
                    />
                    <h3 className="items-title">Responsive Layout</h3>
                    <p className="items-text">Site web accessible sur tablette.</p>
                </div>

                <div className="Why_items-elt" data-testid="items">
                    <div className="image_container">
             
                    </div>
                    <StaticImage
                        alt="Smartphone responsive"
                        src='../images/makbook.png'
                    />
                    <h3 className="items-title">React.js</h3>
                    <p className="items-text">Vous avez un gros projet à venir ? React.js est utilisé par Facebook, Instagram, Twitter, Netflix ...</p>
                </div>
            </div>

            <div className="quote">
                <div className="quote_background"></div>
                </div>
        </div>
    )
}
