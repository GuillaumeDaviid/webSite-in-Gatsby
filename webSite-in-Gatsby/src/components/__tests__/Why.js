import React from "react"
import renderer from "react-test-renderer"
import Why from "../Why.tsx"
import { render, screen } from "@testing-library/react"

describe("Why", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Why />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("should have a title element", () => {
    render(<Why />)

    const title = screen.getByTestId('title');
    expect(title).toBeTruthy();
  })

  it("should have a separator element", () => {
    render(<Why />)

    const separator = screen.getByTestId('separator');
    expect(separator).toBeTruthy();
  })

  it("should have a text element", () => {
    render(<Why />)

    const text = screen.getByTestId('text');
    expect(text).toBeTruthy();
  })

  it("should have 3 items", () => {
    render(<Why />)

    const items = screen.getAllByTestId('items');
    expect(items).toHaveLength(3);
  })
})