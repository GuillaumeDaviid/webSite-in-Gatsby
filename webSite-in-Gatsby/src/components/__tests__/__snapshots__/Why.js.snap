// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Why renders correctly 1`] = `
<div
  className="Why"
>
  <h2
    data-testid="title"
  >
    POURQUOI MOI ?
  </h2>
  <div
    className="Why_separator"
    data-testid="separator"
  />
  <p
    className="Why_head"
    data-testid="text"
  >
    Envie d'un nouveau site web ? Ou alors d'une refonte de celui-ci ? Vous cherchez un développeur qui répond à vos besoins sur-mesure et qui respect les délais et le budget ?
  </p>
  <div
    className="Why_items"
  >
    <div
      className="Why_items-elt"
      data-testid="items"
    >
      <div
        className="image_container"
      />
      <h5
        className="items-title"
      >
        Responsive Design
      </h5>
      <p
        className="items-text"
      >
        Site web accessible sur mobile.
      </p>
    </div>
    <div
      className="Why_items-elt"
      data-testid="items"
    >
      <div
        className="image_container"
      />
      <h5
        className="items-title"
      >
        Responsive Layout
      </h5>
      <p
        className="items-text"
      >
        Site web accessible sur tablette.
      </p>
    </div>
    <div
      className="Why_items-elt"
      data-testid="items"
    >
      <div
        className="image_container"
      />
      <h5
        className="items-title"
      >
        React.js
      </h5>
      <p
        className="items-text"
      >
        Vous avez un gros projet à venir ? React.js est utilisé par Facebook, Instagram, Twitter, Netflix ...
      </p>
    </div>
  </div>
  <div
    className="quote"
  >
    <div
      className="quote_background"
    />
  </div>
</div>
`;
